packages/util-arn-parser/src/index.ts

```diff
@@ -0,0 +1,47 @@
+export interface ARN {
+  partition: string;
+  service: string;
+  region: string;
+  accountId: string;
+  resource: string;
+}
+/**
+ * Validate whether a string is an ARN.
+ */
+export const validate = (str: any): boolean =>
+  typeof str === "string" && str.indexOf("arn:") === 0 && str.split(":").length >= 6;
+
+/**
+ * Parse an ARN string into structure with partition, service, region, accountId and resource values
+ */
+export const parse = (arn: string): ARN => {
+  const matched = arn.split(":");
```

Comment:
If you have a validate function but you don't use it in parse
this forces customers to validate it for themselves.
Why not something like:
```javascript
const [ 
  arnLiteral,
  partition,
  service,
  region,
  accountId,
  ...resource
] = arn.split(":")

// https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
assert(arnLiteral === 'arn', 'Malformed arn')

return {
  partition,
  service,
  region,
  accountId,
  resource: resource.join(":"),
};
```

Reply:
Added the validation to the parser in 684637a